### PR TITLE
Add docker and docker-compose with instructions for people that want to deploy that way.

### PR DESCRIPTION
**Pull request by HendrikPetertje**
Its a fairly simple setup

### CODE REVIEW

**Comment by HendrikPetertje** [comment]
I could probably run a start.sh script that fetches the domain (and other settings) from the system environment and then inject those when people do `docker start` instead of docker-compose commands. That way the Dockerfile itself could have a CMD.

**Comment by HendrikPetertje** [comment]
included that now

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      git \
+      sqlite \
+      sqlite-dev
+
+RUN mkdir /app
+WORKDIR /app
+COPY . /app
+
+RUN cd /app/cmd/server && go build
+
```
Comment (ahdinosaur) [inline Dockerfile:15]: do we want to:

```dockerfile
EXPOSE 8008
EXPOSE 3000
```

?

File: docs/deployment.md [hunk 2]
```diff
@@ -21,6 +21,48 @@ After running `sudo dpkg -i go-ssb-room_v1.2.3_Linux_x86_64.deb` pay special att
...
+```
+
+Followed by
+
+```
+docker-compose up
+```
+
+Your database, secrets and other things will be synchronized to a folder in your
+project called "docker-secrets".
```
Comment (ahdinosaur) [inline docs/deployment.md:43]: why `docker-secrets` and not `ssb-go-room-secrets`?

File: docs/deployment.md [hunk 3]
```diff
@@ -21,6 +21,48 @@ After running `sudo dpkg -i go-ssb-room_v1.2.3_Linux_x86_64.deb` pay special att
...
+
+```
+docker-compose exec room sh
+```
+
+Then inside the virtual machine:
+
+```
+cd /app/cmd/insert-user
+go build
```
Comment (ahdinosaur) [inline docs/deployment.md:57]: should we just build this in the Dockerfile or no? was also wondering why we keep all our build dependencies in the Dockerfile after build, but now i see we need them for this.

**Review by ahdinosaur — APPROVED** [review]
in general looks great, especially as a foundation to get started! :watermelon:

File: Dockerfile (hunk 1 above)
Comment (HendrikPetertje) [reply Dockerfile:15]: We should yes, good one!

File: docs/deployment.md (hunk 2 above)
Comment (HendrikPetertje) [reply docs/deployment.md:43]: I can go with ssb-go-room-secrets ;) I'll make some edits in a bit

File: docs/deployment.md (hunk 3 above)
Comment (HendrikPetertje) [reply docs/deployment.md:57]: jep, you mean that you'd want to remove go from the final build... that might not be the worst idea!

**Comment by HendrikPetertje** [comment]
I have now added a bunch of small fixes as @ahdinosaur has pointed out. I choose not to remove go, it seems that the off-the-shelf alpine image does all kinds of things with GO that I didn't have time to retrace. Aditionally Luaandro had a genius idea of making it possible to include ARM support.

Before I dive into that however I would love to get this one merged (so we can build on top of it).

**Comment by cryptix** [comment]
>  I chose not to remove GO,

I take it I'll close #239 then, too?

> Before I dive into that however I would love to get this one merged (so we can build on top of it).

understood.

(as stated previously, all of this is quite opaque to me but I welcome the contributions a lot!)

**Pull request merged**